bugsnag-android-core/src/main/java/com/bugsnag/android/Session.java

```diff
@@ -244,7 +248,7 @@ public void toStream(@NonNull JsonStream writer) throws IOException {
         }
     }
 
-    private void serializeV2Payload(@NonNull JsonStream writer) throws IOException {
+    private void serializeV2V3Payload(@NonNull JsonStream writer) throws IOException {
```

Comment:
Maybe this should simply be named `serializePayload` and `serializeV1Payload` should maybe be renamed `serializeLegacyPayload`?

```suggestion
    private void serializePayload(@NonNull JsonStream writer) throws IOException {
```